server/swap/swap.go

```diff
@@ -1030,44 +1030,60 @@ func (s *Swapper) TxMonitored(user account.AccountID, asset uint32, txid string)
 	checkMatch := func(match *matchTracker) bool {
 		// Both maker and taker statuses must be locked to check redeem status
 		// and swap tx id.
-		match.makerStatus.mtx.RLock()
-		defer match.makerStatus.mtx.RUnlock()
-		match.takerStatus.mtx.RLock()
-		defer match.takerStatus.mtx.RUnlock()
+		makerStatus, takerStatus := match.makerStatus, match.takerStatus
+		makerStatus.mtx.RLock()
+		defer makerStatus.mtx.RUnlock()
+		takerStatus.mtx.RLock()
+		defer takerStatus.mtx.RUnlock()
 
 		// The swap contract of either the maker or taker must correspond to
 		// specified asset to be of interest.
 		switch asset {
-		case match.makerStatus.swapAsset:
-			// Maker's swap transaction is the asset of interest.
+		case makerStatus.swapAsset:
+			// This is the asset of the maker's swap / takers redeem.
+
+			// For maker, check the swap txid.
 			if user == match.Maker.User() {
 				// The swap contract tx is considered monitored until the swap
 				// is complete, regardless of confirms.
-				return match.makerStatus.swap != nil && match.makerStatus.swap.TxID() == txid
+				return makerStatus.swap != nil && makerStatus.swap.TxID() == txid
 			}
 
-			// Taker's redemption transaction is the asset of interest.
-			_, takerRedeemDone := s.redeemStatus(match.makerStatus, match.takerStatus)
-			if !takerRedeemDone && user == match.Taker.User() &&
-				match.takerStatus.redemption.TxID() == txid {
-				return true
+			// For taker, check the redeem txid, but only if it is not complete.
+			_, takerRedeemDone := s.redeemStatus(makerStatus, takerStatus)
+			if takerRedeemDone {
+				// Taker has a redemption that has reached required confirms.
+				return false // so no longer monitored?
 			}
-		case match.takerStatus.swapAsset:
-			// Taker's swap transaction is the asset of interest.
+
+			// Taker either has not redeemed or the redemption has not reached
+			// the required confirmations. If the txid matches the immature
+			// redeem, it is considered monitored.
+			return takerStatus.redemption != nil && takerStatus.redemption.TxID() == txid
```

Comment:
`redeemStatus` and `makerRedeemStatus` were checking `swapStatus.redeemTime.IsZero()` before attempting to dereference `swapStatus.redemption`.  This PR adds a `redemption != nil` before checking `takerStatus.redemption.TxID()`